### PR TITLE
Use patch Rails to 6.0.3.7 and use workaround for actionpack to mitigate CVEs [MENU-1658]

### PR DESCRIPTION
**Pull request by kraila**
Fully updating actionpack itself would require updating rails from `6.0.3.5` to `6.0.4.1`, and even though it's a patch update, that could get hairy. I've opted for now to upgrade only to `6.0.3.7` (patch of a patch level update), which resolves CVE-2021-22904, CVE-2021-22885, and CVE-2021-22902. I'm mitigating the final CVE using the initializer provided as a workaround with the resolution of [CVE-2021-22942](https://github.com/advisories/GHSA-2rqw-v265-jf8c)

### CODE REVIEW

File: config/initializers/action_dispatch.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+# Patches CVE-2021-22904 https://github.com/advisories/GHSA-7wjx-3g7j-8584
```
Comment (sagotsky) [inline config/initializers/action_dispatch.rb:3]: Was this the right link?  I wanted to double check the patch and got something different looking from early May.

**Review by daniel-patterson-ezcater — COMMENTED** [review]
I'm not completely sure what this is fixing or what it does - perhaps a little more in the PR description and/or as a comment in this file might help?

Also, is there a point where this patch becomes obsolete, eg after we upgrade to a specific gem version? If so, is there a way we could make that clear so we remember to remove this? Potentially we could add that as a comment, or even add a spec that checks the current version number of the specific gem and fails if it's too high, reminding us to deleted that spec and this patch.

**Review by sagotsky — COMMENTED** [review]
I agree with the choice to patch rather than update.  

1. Should there be a test?
2. How do we know when to remove the patch?  I wouldn't want this sitting around after we need it, especially if we upgrade past the version that introduces this change.  Without complicating the patch itself, I could imagine a test against the version of action dispatch.  We can assert that once the gem passes a certain version, this file shouldn't exist.

**Review by tristandunn — COMMENTED** [review]
🎱 Sounds like it's not a common opinion, but I'd favor bumping to 6.0.3.7. Avoids having to monkeypatch and remembering to remove it later, plus we get a few bonus security fixes included:

- [`6.0.3.6`](https://rubyonrails.org/2021/3/26/marcel-upgrade-releases) bumps the [marcel](https://github.com/rails/marcel) dependency to `1.0.0`. It's is used for ActiveStorage, which doesn't seem to be used here.
- [`6.0.3.7`](https://github.com/ezcater/avro-schema-registry/pull/180/files) includes four security fixes, including CVE-2021-22904.

File: config/initializers/action_dispatch.rb (hunk 1 above)
Comment (tristandunn) [inline config/initializers/action_dispatch.rb:3]: ❓Worth adding a ticket to upgrade Rails again to 6.0.4 and note that this can be removed then?

**Review by tristandunn — APPROVED** [review]

File: config/initializers/action_dispatch.rb (hunk 1 above)
Comment (kraila) [reply config/initializers/action_dispatch.rb:3]: good call -- also, I think I linked the wrong CVE in the file there, will update

Comment (kraila) [reply config/initializers/action_dispatch.rb]: ticket made here https://ezcater.atlassian.net/browse/CPC-873

Comment (kraila) [reply config/initializers/action_dispatch.rb]: discussed in open pairing -- it was not the right link, but is one of the ones fixed by bumping rails to 6.0.3.7

**Pull request merged**